quic: flaky TestClientCanDialDifferentQUICVersions

https://github.com/libp2p/go-libp2p/actions/runs/3518375979/jobs/5897459184

```
  === RUN   TestClientCanDialDifferentQUICVersions/Client_dials_both_draft_29_and_v1_on_server_that_supports_both
      conn_test.go:57: using a RSA key: QmUKMJBCWBRRHw77Wcy8ZhyRFpn7pjcyQiUQku2i78Kuhb
      conn_test.go:57: using a RSA key: QmNdxpKB2kpnRPUZu7Rc1zNvrWGr5hY6JVjdJEXTMBPtRy
      conn_test.go:753: 
          	Error Trace:	/home/runner/work/go-libp2p/go-libp2p/p2p/transport/quic/conn_test.go:753
          	Error:      	Received unexpected error:
          	            	Application error 0x0
          	Test:       	TestClientCanDialDifferentQUICVersions/Client_dials_both_draft_29_and_v1_on_server_that_supports_both
  --- FAIL: TestClientCanDialDifferentQUICVersions (0.63s)
      --- PASS: TestClientCanDialDifferentQUICVersions/Client_dials_quic-v1_on_a_quic-v1_only_server (0.00s)
      --- FAIL: TestClientCanDialDifferentQUICVersions/Client_dials_both_draft_29_and_v1_on_server_that_supports_both (0.63s)

--- a/p2p/transport/quic/conn_test.go
+++ b/p2p/transport/quic/conn_test.go
@@ -758,7 +758,6 @@ func TestClientCanDialDifferentQUICVersions(t *testing.T) {
 						panic("unexpected version")
 					}
 					require.NoError(t, err)
-					defer conn.Close()
 
 					_, versionConnLocal, err := quicreuse.FromQuicMultiaddr(conn.LocalMultiaddr())
 					require.NoError(t, err)

